### PR TITLE
Register child process as soon as process starts

### LINKED ISSUE #3040: Clean up child language worker processes
If a language worker process started but connecting to the grpc service running in the functions process fails, child process is not terminated when functions host restarts/crashes. This results in lingering tcp/ip ports in [CLOSE_WAIT](https://blogs.technet.microsoft.com/janelewis/2010/03/09/explaining-close_wait/) state.

### PR DESCRIPTION
**Pull request by pragnagopa**
- Fixes #3040 
- Log child process logs written to stderr as errors

### CODE REVIEW

**Comment by pragnagopa** [comment]
@fabiocav - Can you please review?

File: src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs [hunk 1]
```diff
@@ -338,7 +336,14 @@ internal void StartProcess(string workerId, Process process)
                 // TODO: per language stdout/err parser?
                 if (e.Data != null)
                 {
-                    _logger.LogInformation(e.Data);
+                    if (e.Data.ToLower().Contains("warn"))
```
Comment (fabiocav) [inline src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs]: nit: remove `ToLower` and use the `.Contains("warn", StringComparer.OrdinalIgnoreCase)`

File: src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs [hunk 2]
```diff
@@ -365,8 +370,10 @@ internal void StartProcess(string workerId, Process process)
                     HandleWorkerError(new Exception("Worker process is not attached"));
                 }
             };
-            _logger.LogInformation($"Start Process: {process.StartInfo.FileName} {process.StartInfo.Arguments}");
+            _logger.LogInformation($"Starting {process.StartInfo.FileName} language worker process with Arguments={process.StartInfo.Arguments}");
             process.Start();
+            _logger.LogInformation($"{process.StartInfo.FileName} process with Id={process.Id} started");
+            _processRegistry?.Register(_process);
```
Comment (fabiocav) [inline src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs]: This call should probably me moved to `StartWorker`, so the scope of `StartProcess` remains the same (it only starts the process with the appropriate configuration)

**Comment by pragnagopa** [comment]
Opened issue https://github.com/Azure/azure-functions-host/issues/3050 to refactor process management

**Pull request merged**